### PR TITLE
Create index reads and novaseq support

### PR DESCRIPTION
**Pull request by johandahlberg**
**What problems does this PR solve?**
This PR has two things in it:
 - An added option to make it easier to get bcl2fastq to created fastq-files for indexes (which is needed e.g. for Chromium runs).
 - It adds support for NovaSeq, while simultaneously removing the dependency we had on Illuminate, which was a rather heavy download for the few things we used it for.

**How has the changes been tested?**
I have manually confirmed this functionality on a NovaSeq run.

**Reasons for careful code review**
If any of the boxes below are checked, extra careful code review should be inititated.

  - [ ] This PR contains code that could remove data

### CODE REVIEW

File: bcl2fastq/lib/bcl2fastq_utils.py
Comment (b97pla) [inline bcl2fastq/lib/bcl2fastq_utils.py:139]: In the future, I think this mapping would be nice to have in a config..

File: bcl2fastq/lib/bcl2fastq_utils.py
Comment (b97pla) [inline bcl2fastq/lib/bcl2fastq_utils.py:142]: if future machine type prefixes are superstrings/substrings of existing prefixes, we need to remember to match more carefully

File: config/app.config
Comment (b97pla) [inline config/app.config:32]: is this just an example config? otherwise, I believe we're on `2.20.x`

**Review by b97pla — APPROVED** [review]
:+1: Looks good, a few comments but I'm ok to merge when you're happy.

File: bcl2fastq/lib/bcl2fastq_utils.py
Comment (johandahlberg) [reply bcl2fastq/lib/bcl2fastq_utils.py:139]: Yes. I agree. Right now however I think this will have to do.

File: bcl2fastq/lib/bcl2fastq_utils.py
Comment (johandahlberg) [reply bcl2fastq/lib/bcl2fastq_utils.py:142]: Yes. However at the moment they are not, and fingers crossed, Illumina will keep the prefixes unique (or provide some better way of identifying the machine type).

File: config/app.config
Comment (johandahlberg) [reply config/app.config:32]: Yes. This is just an example config. One can specify the paths to multiple versions in the config as well, so I'm not sure we need to keep this particular one up to date.

**Comment by johandahlberg** [comment]
I agree with all your comments, but since this is a side-track to me fixing other things I'd like to keep it as is for now. Wait just a little bit before merging though, so I can bring more changes in if I should identify some last minute problems while I keep working on getting the CI stuff working.

**Comment by johandahlberg** [comment]
This is ready to merge now.

**Pull request merged**
